runtimes/runtimes/util/pathUtil.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Simplified version of path.join that can be safely used on web
+ * @param segments
+ * @returns
+ */
+export function joinUnixPaths(...segments: string[]): string {
+    // Filter out empty segments and normalize each segment
+    const normalizedSegments = segments.filter(Boolean).map(segment => segment.replace(/^\/+|\/+$/g, ''))
```

Comment:
can also replace `\` slashes with `/` for Windows. then the following logic works on all platforms